go/vt/vtgate/planbuilder/horizon_planning.go

```diff
@@ -145,19 +145,61 @@ func (hp *horizonPlanning) planAggregations() error {
 	}
 
 	for _, e := range hp.qp.SelectExprs {
-		offset, _, err := pushProjection(e.Col, hp.plan, hp.semTable, true, false)
-		if err != nil {
-			return err
+		// push all expression if they are non-aggregating or the plan is not ordered aggregated plan.
+		if !e.Aggr || oa == nil {
+			_, _, err := pushProjection(e.Col, hp.plan, hp.semTable, true, false)
+			if err != nil {
+				return err
+			}
+			continue
 		}
+
 		if e.Aggr && oa != nil {
```

Comment:
should this if not be removed altogether? we should always do this, if the previous `if` wasn't executed, right?